lib/helpers/buildURL.js

```diff
@@ -35,12 +35,20 @@ module.exports = function buildURL(url, params, options) {
 
   var _encode = options && options.encode || encode;
 
-  var serializerParams = utils.isURLSearchParams(params) ?
-    params.toString() :
-    new AxiosURLSearchParams(params, options).toString(_encode);
+  var serializeFn = options && options.serialize;
 
-  if (serializerParams) {
-    url += (url.indexOf('?') === -1 ? '?' : '&') + serializerParams;
+  var serializedParams;
+
+  if (serializeFn) {
+    serializedParams = serializeFn(params, options);
+  } else {
+    serializedParams = utils.isURLSearchParams(params) ?
+      params.toString() :
+      new AxiosURLSearchParams(params, options).toString(_encode);
+  }
+
+  if (serializedParams) {
```

Comment:
@JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in v1.x here: https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js

Reply:
@JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in v1.x here: 

https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js#L39-L57

Reply:
@JadeMugs I disagree. I don't think it matters much since the `hashMark` part of the code is only to strip out the URL. Besides, I'm following the version in https://github.com/axios/axios/pull/5113 here: 

https://github.com/DigitalBrainJS/axios/blob/92f0758429541b05c614dc25bc712379d8a41dbd/lib/helpers/buildURL.js#L39-L57